pkg/kubelet/prober/worker.go

```diff
@@ -316,6 +325,29 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		return true
 	}
 
+	if resultChanged {
```

Comment:
@intUnderflow can you check how this will work with liveness probes? Will we get two events - one just before container is killed and another - after it got restarted?

Reply:
There's no report of the probe changing to status Success. Only failure.

Since the Pod is recreated on failure a new probe manager is created, which causes there to be a new worker, which resets the lastResult to value 0, 0 maps to results.Success which means when the probe passes in the future theres no report.

I used the following deployment
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test-deployment
  labels:
    app: test-deployment
spec:
  replicas: 3
  selector:
    matchLabels:
      app: test-deployment
  template:
    metadata:
      labels:
        app: test-deployment
    spec:
      containers:
      - name: test-deployment
        image: uber/k8s-fail-server-one-minute:latest
        ports:
        - containerPort: 80
        livenessProbe:
          httpGet:
            path: /
            port: 80
          initialDelaySeconds: 3
          periodSeconds: 3
```

This to me feels like the expected and good behaviour